src/main.zig

```diff
@@ -66,11 +70,42 @@ fn handleBuiltin(argv: [][]const u8, ally: *std.mem.Allocator) !bool {
             std.process.exit(exit_num);
         }
     }
+
     if (std.mem.eql(u8, argv[0], "cd")) {
-        if (argv.len > 2) {
+        var operands = try Argv.initCapacity(ally, 1);
+        defer operands.deinit();
+        var flags = Flags.init(ally);
+        defer flags.deinit();
+        for (argv[1..]) |a| {
+            if (a[0] != '-') { // without a dash it's an operand
+                try operands.append(a);
+            } else { // TODO: handle path starting with '/', '.' or neither(CDPATH) explicitely
+                if (a.len == 1) { // singular dash means go back to previous directory
+                    const newdir = try std.process.getEnvVarOwned(ally, "OLDPWD");
+                    defer ally.free(newdir);
+                    try stdout.print("{s}\n", .{newdir});
+                    try cd(ally, newdir);
+                    return true;
+                } else { // Otherwise it's a flag
+                    for (a[1..]) |flag| {
+                        switch (flag) {
+                            'L' => try flags.append(.L),
+                            'P' => try flags.append(.P),
+                            else => {
+                                try shigError("cd: Illegal option -{c}", .{flag});
+                                return true;
+                            },
+                        }
+                    }
+                    // TODO: make the flags do sth
```

Comment:
Please do not introduce the flags if you do not make them do something.

Reply:
fair, I just thought the end goal was posix compliance. But might as well focus on the stuff I actually use first

Reply:
Yep, right now, the goal is a working shell and good code. The end goal is posix compliance :)

Reply:
flags are gone for now